code/modules/research/nanites/nanite_program_hub.dm

```diff
@@ -26,6 +26,14 @@
 	. = ..()
 	linked_techweb = SSresearch.science_tech
 
+/obj/machinery/nanite_program_hub/update_icon()
+	. = ..()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+	if((machine_stat & (NOPOWER|MAINT|BROKEN)) || panel_open)
+		return
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_program_hub_on", layer, plane)
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_program_hub_on", EMISSIVE_LAYER, EMISSIVE_PLANE)
```

Comment:
why aren't you doing this via update_overlays?

Reply:
my bad, i initially did, it didn't work, changed it to update_icons to try to fix it, and forgot to switch it back after i found the actual issue